### ISSUE TITLE
It just doesn't calculate the metrics

### ISSUE THREAD

**Issue by xiehuanyi** [huggingface/peft#782]
### System Info

Package                  Version
------------------------ ------------
accelerate               0.21.0
evaluate                 0.4.0
huggingface-hub          0.13.2
tokenizers               0.13.2
torch                    2.0.1
torch-geometric          2.3.1
torchaudio               0.13.1
torchmd-net              0.2.6
torchmetrics             0.11.0
torchvision              0.14.1
transformers             4.27.1

### Who can help?

@pacman100 @younesbelkada @sayakpaul 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```
model = OPTForSequenceClassification.from_pretrained(
    model_dir, 
    problem_type='regression', 
    num_labels=2
    )

tokenizer = AutoTokenizer.from_pretrained(model_dir)
peft_config = LoraConfig(
    TaskType.SEQ_CLS, 
    inference_mode=False, 
    r=8,
    lora_alpha=32, 
    lora_dropout=0.1)

model = get_peft_model(model, peft_config)
model.print_trainable_parameters()
training_steps = epochs * len(train_set) // bs
warmup_steps = int(0.1 * training_steps)

opt = Lion(model.parameters(), lr=1e-4)
scheduler = get_linear_schedule_with_warmup(opt, num_warmup_steps=warmup_steps, num_training_steps=training_steps)

early_stop = EarlyStoppingCallback(early_stopping_patience=3)
def compute_metrics(eval_pred):
    preds, labels = eval_pred
    col_rmse = np.sqrt(np.mean((preds - labels) ** 2, axis=0))
    mcrmse = np.mean(col_rmse)
    res = {
        "content_rmse": col_rmse[0],
        "wording_rmse": col_rmse[1],
        "eval_mcrmse": mcrmse,
    }
    return res

args = TrainingArguments(
    output_dir=output_dir, 
    do_train=True, 
    do_eval=True, 
    num_train_epochs=epochs,
    lr_scheduler_type='linear', 
    per_device_train_batch_size=bs,
    gradient_accumulation_steps=4,
    logging_steps=8,
    warmup_ratio=0.1, 
    save_total_limit=1, 
    fp16=True,
    save_strategy='steps',
    evaluation_strategy='steps',
    save_steps=100,
    eval_steps=10
)

trainer = Trainer(
    model=model, 
    args=args, 
    data_collator=collate_fn, 
    train_dataset=train_set, 
    eval_dataset=eval_set,
    tokenizer=tokenizer,
    optimizers=(opt, scheduler),
    compute_metrics=compute_metrics
)
trainer.train()
```
here are my full code, you can download the dataset from kaggle
```
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from transformers import AutoTokenizer, OPTForSequenceClassification, get_linear_schedule_with_warmup, \
    EarlyStoppingCallback
from peft import LoraConfig, TaskType, get_peft_model
from torch.utils.data import Dataset
from transformers import Trainer, TrainingArguments
import torch
from lion_pytorch import Lion
epochs = 3
bs = 1
def seed_everything(seed: int):
    import random, os
    import numpy as np
    import torch
    
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = True
    
seed_everything(42)
dataset_dir = '/data/kaggle'
model_dir = '/opt-125m'
output_dir = '/commonlit/opt-125m'
print(f"dataset: {os.listdir(dataset_dir)}\n")
print(f"models: {os.listdir(model_dir)}")
df_prompts = pd.read_csv(os.path.join(dataset_dir, 'prompts_train.csv'))
df_prompts
df_summaries = pd.read_csv(os.path.join(dataset_dir, 'summaries_train.csv'))

df_summaries.head(5)
df_merged = pd.merge(df_summaries, df_prompts[['prompt_id', 'prompt_question', 'prompt_title', 'prompt_text']], on='prompt_id')
df_merged.head(6)
def template(inputs):
    context = f"Predict the content and wording scores of the summary based on the following words:\n\n" + \
        f"Prompt question: {inputs['prompt_question']}\n" + \
        f"Prompt title: {inputs['prompt_title']}\n" + \
        f"Prompt text: {inputs['prompt_text']}\n\n" + \
        f"Summary: {inputs['text']}\n\n"
    return context
        
    
print(template(df_merged.iloc[0]))
model = OPTForSequenceClassification.from_pretrained(
    model_dir, 
    problem_type='regression', 
    num_labels=2
    )

tokenizer = AutoTokenizer.from_pretrained(model_dir)
peft_config = LoraConfig(
    TaskType.SEQ_CLS, 
    inference_mode=False, 
    r=8,
    lora_alpha=32, 
    lora_dropout=0.1)

model = get_peft_model(model, peft_config)
model.print_trainable_parameters()
class SummariesSet(Dataset):
    def __init__(self, df) -> None:
        super().__init__()
        self.df = df
    
    def __getitem__(self, idx) -> str:
        sample = self.df.iloc[idx]
        return template(sample), sample['content'], sample['wording']
    
    def __len__(self):
        return len(self.df)

def collate_fn(batch):
    labels = np.array([[i[1], i[2]] for i in batch])
    batch = [i[0] for i in batch]
    batch = tokenizer(batch, max_length=1024, padding=True, truncation=True, return_tensors='pt')
    batch['labels'] = torch.tensor(labels, dtype=torch.float16)
    return batch

data_len = len(df_merged)
train_eval_split = 0.9
train_size = int(train_eval_split * data_len)
eval_size = data_len - train_size

df_merged = df_merged.sample(frac=1)
train_set = SummariesSet(df_merged.head(train_size))
eval_set = SummariesSet(df_merged.tail(eval_size))

print('='*10, 'test', '=' * 10)
print(len(train_set), len(eval_set))
batch = [train_set[i] for i in range(3)]
print(collate_fn(batch))
training_steps = epochs * len(train_set) // bs
warmup_steps = int(0.1 * training_steps)

opt = Lion(model.parameters(), lr=1e-4)
scheduler = get_linear_schedule_with_warmup(opt, num_warmup_steps=warmup_steps, num_training_steps=training_steps)

early_stop = EarlyStoppingCallback(early_stopping_patience=3)
def compute_metrics(eval_pred):
    preds, labels = eval_pred
    col_rmse = np.sqrt(np.mean((preds - labels) ** 2, axis=0))
    mcrmse = np.mean(col_rmse)
    res = {
        "content_rmse": col_rmse[0],
        "wording_rmse": col_rmse[1],
        "eval_mcrmse": mcrmse,
    }
    return res

args = TrainingArguments(
    output_dir=output_dir, 
    do_train=True, 
    do_eval=True, 
    num_train_epochs=epochs,
    lr_scheduler_type='linear', 
    per_device_train_batch_size=bs,
    gradient_accumulation_steps=4,
    logging_steps=8,
    warmup_ratio=0.1, 
    save_total_limit=1, 
    fp16=True,
    save_strategy='steps',
    evaluation_strategy='steps',
    save_steps=100,
    eval_steps=10
)

trainer = Trainer(
    model=model, 
    args=args, 
    data_collator=collate_fn, 
    train_dataset=train_set, 
    eval_dataset=eval_set,
    tokenizer=tokenizer,
    optimizers=(opt, scheduler),
    compute_metrics=compute_metrics
)
trainer.train()
```

### Expected behavior

When I use peft lora, it just doesn't evalutate the eval_set. if I use lora, it will output this
<img width="716" alt="截屏2023-08-03 18 12 35" src="https://github.com/huggingface/peft/assets/74809554/961d2654-8078-436e-afb1-89cefd47ac71">
but if i comment it, use ordinary transformer, it works fine
<img width="706" alt="截屏2023-08-03 18 14 05" src="https://github.com/huggingface/peft/assets/74809554/2ac513fa-6bed-4821-89dc-f07a4511028b">

**Comment by nbroad1881** [comment]
Try setting `label_names="labels"` in `TrainingArguments`
